karate-tests/src/test/java/helpers/utils.feature

```diff
@@ -1,45 +1,89 @@
 @ignore
 Feature: Utils
 
-A collection of utility functions...
+A collection of utility functions for generating, validating and manipulating data
 
 Scenario:
-    * def randomUUID = function(){ return java.util.UUID.randomUUID() + '' }
-    * def randomInt = function() { return Math.random() * (10000 - 1) + 1 }
-
-    * def isTodaysDate = 
-    """
-    function(inputDate) {
-      var input = (new Date(Date.parse(inputDate)).toDateString());
-      var today = (new Date()).toDateString();
-      return input == today;
-    }
-    """
-    
-    * def getRandomBirthDate = 
-    """
-    function() {
-      var min = new Date(Date.parse('1930-01-01')).getTime();
-      var max = new Date().getTime();
-      var randomValue = Math.random() * (max - min) + min;
-
-      var randomDate = new Date(randomValue)
-      return randomDate.toISOString().split('T')[0]
-    }
-    """
+  # data generators
+  * def randomUUID = function(){ return java.util.UUID.randomUUID() + '' }
+  * def randomInt = function() { return Math.floor(Math.random() * 999) + 1 }
+  * def randomString = 
+  """
+  function(length) {
+    let result = '';
+    const characters = 'abcdefghijklmnopqrstuvwxyz';
+    const charactersLength = characters.length;
 
-    * def sleep = function(pause){ java.lang.Thread.sleep(pause) }
-    
-    # pick an item at random from an array, ignoring one of the items
-    * def pickDifferentOption = 
-    """
-    function(optionsArray, itemToIgnore) {
-      const index = optionsArray.indexOf(itemToIgnore);
-      optionsArray.splice(index, 1)
-      return optionsArray[Math.floor(Math.random() * optionsArray.length)]
+    for (let i = 0; i < length; i++) {
+        result += characters.charAt(Math.floor(Math.random() * charactersLength));
     }
-    """
+
+    return result;
+  }
+  """
+  * def randomBirthDate = 
+  """
+  function() {
+    var min = new Date(Date.parse('1930-01-01')).getTime();
+    var max = new Date(Date.parse('2023-09-01')).getTime();
+    var randomValue = Math.random() * (max - min) + min;
+
+    var randomDate = new Date(randomValue)
+    return randomDate.toISOString().split('T')[0]
+  }
+  """
+
+  # data validators
+  * def isTodaysDate = 
+  """
+  function(inputDate) {
+    var input = (new Date(Date.parse(inputDate)).toDateString());
+    var today = (new Date()).toDateString();
+    return input == today;
+  }
+  """
+
+  * def isValidTimestamp = function(timestamp) { return !isNaN(Date.parse(timestamp)) }
+  * def isValidDateString = 
+  """
+  function (dateString) {
+    var regex=new RegExp("([0-9]{4}[-](0[1-9]|1[0-2])[-]([0-2]{1}[0-9]{1}|3[0-1]{1})|([0-2]{1}[0-9]{1}|3[0-1]{1})[-](0[1-9]|1[0-2])[-][0-9]{4})");
```

Comment:
I'm guessing the regex tests pass, seems a very lengthy regex for a uk postcode, not saying it's wrong though - just surprised it's so long, same applies to date regex

Reply:
@richardseacome1-nhs I've had a lot of "fun" trying to find a good regex for uk postal codes... I can't actually find one that is 100% trustworthy. But yes, they are typically long.